### PR TITLE
Create eight-of-the-biggest-lies-in-APIs

### PR DESCRIPTION
**Pull request by MatthewReinbold**
Submitting an article for the APIs You Won't Hate blog. Will need a cover image. In addition, @philsturgeon thought it might be worthwhile to springboard from this piece and link to other, existing articles (for example, could see the cache lie easily linking to the published work on caching). Somebody with knowledge of the deep cuts will need to add those, if desired.

### CODE REVIEW

**Comment by mbifulco** [comment]
Thanks @MatthewReinbold ! I opened #582 for this article as well when you brought it up in Slack..  Can you add an author bio file to this PR? It should go in `/src/content/authors/matthew-reinbold.mdx`, and you can use [Phil's](https://raw.githubusercontent.com/apisyouwonthate/apisyouwonthate.com/main/src/content/authors/phil-sturgeon.mdx) as a template. Feel free to leave twitter/instagram/etc blank as you see fit.

File: src/content/blog/eight-of-the-biggest-lies-in-APIs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+subtitle: >-
+  In the API space, numerous statements are context-dependent, misunderstandings, or outright lies. Here are some of the biggest the APIs You Won't Hate community has heard to date.
+date: "2022-01-12"
+coverImage: 
+type: blog
+author: Matthew Reinbold
+---
+Have you ever been told "Everybody does it this way" when arguing about API design? Or have you heard, "We use HTTPS, so our API is secure"? 
+
+In the API space, numerous statements are highly context-dependent, misunderstandings, or outright lies. To find some of the most egregious of these, I asked the APIs You Won't Hate Slack for those phases that stand their neck hair on end and their eyes to see red. If you find yourself about to use one of these jargon phrases unironically in your next architectural debate, please think twice.
```
Comment (philsturgeon) [inline src/content/blog/eight-of-the-biggest-lies-in-APIs:12]:
```suggestion
In the API space, numerous statements are highly context-dependent, misunderstandings, or outright lies. To find some of the most egregious of these, I asked the [APIs You Won't Hate Slack Community](https://slack.apisyouwonthate.com/) for the phases that stand their neck hair on end, and make their eyes go red. If you find yourself about to use one of these jargon phrases unironically in your next architectural debate, please think twice. 
```

File: src/content/blog/eight-of-the-biggest-lies-in-APIs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+author: Matthew Reinbold
+---
+Have you ever been told "Everybody does it this way" when arguing about API design? Or have you heard, "We use HTTPS, so our API is secure"? 
+
+In the API space, numerous statements are highly context-dependent, misunderstandings, or outright lies. To find some of the most egregious of these, I asked the APIs You Won't Hate Slack for those phases that stand their neck hair on end and their eyes to see red. If you find yourself about to use one of these jargon phrases unironically in your next architectural debate, please think twice. 
+
+## The Eight Greatest API Lies
+### #8: "It's just a PoC/MVC/Experiment; we'll tidy up the design once we have feedback." (Frank Kilcommins)
+
+Iterative development is a beautiful thing and a cornerstone of modern software processes. The problem is applying the same approach to an API design. Unlike code, an interface is a promise between one and many individuals. And no matter how an API producer might imply that the relationship has no strings attached, breaking promises undermines trust and hurts feelings.
```
Comment (philsturgeon) [inline src/content/blog/eight-of-the-biggest-lies-in-APIs:17]:
```suggestion
Iterative development is a beautiful thing, and a cornerstone of modern software processes. The problem is applying the same approach to an API design. Unlike code, an interface is a promise between one and many individuals. And no matter how an API producer might imply that the relationship has no strings attached, breaking promises undermines trust, and can create incredible expense as time and money is wasted on avoidable rewrites.
```

File: src/content/blog/eight-of-the-biggest-lies-in-APIs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## The Eight Greatest API Lies
+### #8: "It's just a PoC/MVC/Experiment; we'll tidy up the design once we have feedback." (Frank Kilcommins)
+
+Iterative development is a beautiful thing and a cornerstone of modern software processes. The problem is applying the same approach to an API design. Unlike code, an interface is a promise between one and many individuals. And no matter how an API producer might imply that the relationship has no strings attached, breaking promises undermines trust and hurts feelings. 
+
+It is tough to get otherwise busy developers to spend their precious time creating code for an interface *the first time*. Imagine their excitement discovering their work was invalidated and that they could now re-implement what was previously working? Long story short, most don't. 
+
+### #7: "It's internal and on on-prem, so we don't need to worry about Auth for a V1." (Frank Kilcommins)
```
Comment (philsturgeon) [inline src/content/blog/eight-of-the-biggest-lies-in-APIs:21]:
```suggestion
### #7: "It's internal and on-prem, so we don't need to worry about Auth for a V1." (Frank Kilcommins)
```

File: src/content/blog/eight-of-the-biggest-lies-in-APIs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+> "Caching is the ability to store copies of frequently accessed data in several places along the request-response path."
+
+Leveraging caches improve the API experience by:
+
+* Reducing bandwidth
+* Reducing latency
+* Reducing server load
+* Bridging network failures
+
+To ignore caching's usefulness to an API is like eschewing indoor plumbing; sure, you can make things work, but at what cost?
```
Comment (philsturgeon) [inline src/content/blog/eight-of-the-biggest-lies-in-APIs:42]:
```suggestion
To ignore caching's usefulness to an API and [it's clients](/blog/speeding-up-apis-apps-smart-toasters-with-http-response-caching) is like eschewing indoor plumbing; sure, you can make things work, but at what cost? 
```

**Review by philsturgeon — COMMENTED** [review]
Looking good with a few lil tweaks here and there! Thank you for this.

**Comment by philsturgeon** [comment]
Which stock image is best (or anything else from istock).

image

image
image

**Review by MatthewReinbold — COMMENTED** [review]
Looks good. I resolved the changes.

**Comment by MatthewReinbold** [comment]
That "fake/fact" one is pretty clever. 

I also briefly noticed a typo that slipped past grammarly - in the first couple paragraphs, I referred to "phase" instead of "phrase". That should be corrected before going out.

**Comment by philsturgeon** [comment]
Lgtm!

**Review by philsturgeon — DISMISSED** [review]

**Comment by mbifulco** [comment]
ty @philsturgeon  - I'm going to commit a few changes to tidy up titles in the article. I'll get this published by the end of the weekend.

**Comment by philsturgeon** [comment]
Great! Let’s make sure we got the actual istock image I was a bit worried we just used the one I pasted but that might just have been tired eyes / mobile.

**Review by mbifulco — APPROVED** [review]

**Pull request merged**